pkg/reconciler/reconcile.go

```diff
@@ -0,0 +1,23 @@
+package reconciler
+
+import (
+	"context"
+
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/db"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+
+	"go.uber.org/zap"
+)
+
+func Reconcile(logger *zap.Logger, client *db.Database, req types.Request) types.Result {
```

Comment:
Please pass the logger around via the logr.Logger interface (as requested in previous reviews)

Reply:
AFAIK The above is not possible because zap.Logger and logr.Logger are not compatible to each other as we do in reconciler in other project

Reply:
Why? this is exactly the strategy operator framework is taking for generated operators:
https://github.com/red-hat-storage/managed-fusion-agent/blob/main/main.go#L103

Reply:
This is not possible as discussed we need to write wrapper over zap for this one, will take care of it in followup PR

Reply:
Ok, as you already clarified offline. resolving the comment.